react-app/src/native/components/register/ChooseLeague.js

```diff
@@ -0,0 +1,105 @@
+import React, {useState} from 'react'
+import PropTypes from 'prop-types' 
+import { connect } from 'react-redux'
+import {Picker, View, Icon } from 'native-base'
+import { fetchLeague } from '../../../actions/leagues' //Gets the leagues from the web.
+import DateTimeHelpers from '../../../utils/datetimehelpers'
+
+//Expects a sport id given as 1-4 under the 'sport' props tag
+
+class PickLeagues extends React.Component { 
+  
+  //I make a server call to get all the leagues. In the props (sport=(1-4)) the correct leagues for that sport appear
+  static propTypes = {
+    seasons: PropTypes.object.isRequired,
+    isLoading: PropTypes.bool.isRequired,
+    leagues: PropTypes.object.isRequired,
+    getLeague: PropTypes.func.isRequired 
+  }
+  
+  constructor(props) {
+    super(props)
+    state = {
+      counter:[1]
+    }
+  }
+
+  render() {
+    
+    const {
+      loading,
```

Comment:
You're not showing a loading screen if the page is still loading data. If that's not a thing for this page for whatever reason (all the data is preloaded before you can get here) then remove all references to the prop. I'd prefer you show the loading page if the data is loading, that way if we change how this component is used it won't just break by accident.